### DISCUSSION TITLE
Search dropdown: is it possible to "set query"?

### DISCUSSION THREAD

**Discussion by cheesyMan** [fomantic/Fomantic-UI#2095] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Let's suppose I've a search selection dropdown with fruit:

- Apple
- Grapes
- Lemon
- Orange
- Peach
- Pear
- Pineapple
- Pink Grapefruit
- Red Grapefruit

Id like to do something like` $("#fruitList").dropdown("set query", "grape")` to programatically insert a search term and see a filtered list:

- **Grape**s
- Pink **Grape**fruit
- Red **Grape**fruit

as a result.

I know a "set query" behavior is not present in dropdown; is there a way to achieve the same result in any way?

**Comment by lubber-de** [comment, marked as answer]
You can achieve this by adding the search value manually .
```
$('.ui.dropdown > input.search').val('grape').focus(); //if you do not focus, the dropdown field will not open
$('.ui.dropdown > .text').addClass('filtered');	// needed to remove the placeholder
```

See https://jsfiddle.net/lubber/dn4ry57a/21/

**Comment by cheesyMan** [comment]
Hi @lubber-de thanks for your suggestion.

Unfortunately it looks like it doesn't work with my dropdown settings.

```
$('.ui.dropdown').dropdown({
	showOnFocus: false,
	fullTextSearch: true,
	match: "text",
	forceSelection: false,
});
```

Maybe it's for the **showOnFocus** setting?

Anyway, also adding a `$('ui.dropdown').dropdown('show')` line to suggested code, seems useless, showing the dropdown BUT not filtered.
(see https://jsfiddle.net/72f1tbhd/)

**Reply by lubber-de** [reply]
In your use case add the `filter` behavior infront of the `show` behavior:
See https://jsfiddle.net/lubber/enqg34ho/

**Reply by cheesyMan** [reply]
perfect! thanks
